src/main/java/com/uid2/admin/vertx/service/SharingService.java

```diff
@@ -106,15 +118,46 @@ private void handleSetKeyset(RoutingContext rc) {
                 }
             } else {
                 keysetId = Collections.max(keysetsById.keySet()) + 1;
+                create = true;
```

Comment:
Can't we just check here if the site id already has a keyset and return an error?

Reply:
relocated the logic